nifi-nar-bundles/nifi-cdc/nifi-cdc-postgresql-bundle/nifi-cdc-postgresql-processors/pom.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements. See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License. You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.nifi</groupId>
+        <artifactId>nifi-cdc-postgresql-bundle</artifactId>
+        <version>1.16.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>nifi-cdc-postgresql-processors</artifactId>
+    <packaging>jar</packaging>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-api</artifactId>
+            <version>1.16.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-cdc-api</artifactId>
+            <version>1.16.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-utils</artifactId>
+            <version>1.16.0-SNAPSHOT</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.nifi</groupId>
+            <artifactId>nifi-mock</artifactId>
+            <version>1.16.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-simple</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/org.postgresql/postgresql -->
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.3.1</version>
```

Comment:
What is the license for postgres lib and what transitive deps does this pull in?  We need to account for these in the License and Notice as appropriate.  If this part is easy it means it hasn't been done yet.

Reply:
The PostgreSQL JDBC Driver is distributed under the BSD-2-Clause License. The simplest explanation of the licensing terms is that you can do whatever you want with the product and source code as long as you don't claim you wrote it or sue us. You should give it a read though, it's only half a page.

Copyright (c) 1997, PostgreSQL Global Development Group
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

1. Redistributions of source code must retain the above copyright notice,
   this list of conditions and the following disclaimer.
2. Redistributions in binary form must reproduce the above copyright notice,
   this list of conditions and the following disclaimer in the documentation
   and/or other materials provided with the distribution.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
POSSIBILITY OF SUCH DAMAGE.

Reply:
Looks like it is https://github.com/pgjdbc/pgjdbc/blob/master/LICENSE. We'll need that in the LICENSE for this nar file (see other nars for examples) and we'll need this in the LICENSE of the nifi-assembly.   We don't need it in the top level LICENSE of the entire source tree since this is a binary only dependency.  There are these two immediate dependencies

<groupId>org.checkerframework</groupId>
<artifactId>checker-qual</artifactId>
<version>3.5.0</version>
<scope>runtime</scope>
</dependency>
<dependency>
<groupId>com.github.waffle</groupId>
<artifactId>waffle-jna</artifactId>
<version>1.9.1</version>
<optional>true</optional>

Both appear to be MIT so they too would go in the mentioned LICENSES.  But jna appears to have several deps that also need to be followed up on.

Reply:
@gerdansantos Yep I know.  I am referring to what it takes to actually use these things in Apache Software Foundation releases.  We need proper license/notice reflections for this.  We have it documented here https://nifi.apache.org/licensing-guide.html.  Take a look at my previous reply.  And find a NOTICE/LICENSE pair in another nar as an example of how we take care of the nar.  The nar LICENSE/NOTICE should reflect precisely what libs are bundled within it.  The LICENSE of the nifi-assembly needs anything not already mentioned added to it for anything that is category A.

This part is admittedly the worst part and the biggest shocker for folks who want to contribute.  Happy to help here but there is legwork.

Reply:
@joewitt No problem man! We'll take a look and and do whatever is necessary to comply with this license issue.
Thank you so much for the feedback and tips!

Reply:
Hey @joewitt, the waffle-jna is not present in mvn dependency tree. Maybe because it is an optional dependency for postgresql lib. Take a look:

$ mvn dependency:tree
...
--- maven-dependency-plugin:3.2.0:tree (default-cli) @ nifi-cdc-postgresql-nar ---
org.apache.nifi:nifi-cdc-postgresql-nar:nar:1.16.0-SNAPSHOT
+- org.apache.nifi:nifi-cdc-postgresql-processors:jar:1.16.0-SNAPSHOT:compile
|  +- org.apache.nifi:nifi-cdc-api:jar:1.16.0-SNAPSHOT:compile
|  |  \- org.apache.commons:commons-lang3:jar:3.9:compile
|  +- org.apache.nifi:nifi-utils:jar:1.16.0-SNAPSHOT:compile
|  |  +- org.apache.nifi:nifi-api:jar:1.16.0-SNAPSHOT:provided
|  |  +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:compile
|  |  |  \- jakarta.activation:jakarta.activation-api:jar:1.2.2:compile
|  |  \- org.glassfish.jaxb:jaxb-runtime:jar:2.3.5:compile
|  |     +- org.glassfish.jaxb:txw2:jar:2.3.5:compile
|  |     +- com.sun.istack:istack-commons-runtime:jar:3.0.12:compile
|  |     \- com.sun.activation:jakarta.activation:jar:1.2.2:runtime
**|  +- org.postgresql:postgresql:jar:42.3.3:compile
|  |  \- org.checkerframework:checker-qual:jar:3.5.0:runtime**
|  \- com.fasterxml.jackson.core:jackson-databind:jar:2.12.5:compile
|     +- com.fasterxml.jackson.core:jackson-annotations:jar:2.12.5:compile
|     \- com.fasterxml.jackson.core:jackson-core:jar:2.12.5:compile  
...

In this case, I added only the postgresql and checker-qual licenses. Is this okay?